CMakeLists.txt

```diff
@@ -0,0 +1,201 @@
+cmake_minimum_required(VERSION 3.14)
+project(mold VERSION 1.3.0)
+
+include(CMakeDependentOption)
+
+# FIXME: this is for parity with the makefiles that install directly to <prefix>/lib
+# NOTE: defining this before including GNUInstallDirs makes its platform detection a noop
+set(CMAKE_INSTALL_LIBDIR lib CACHE PATH "")
+include(GNUInstallDirs)
+
+# Mold build options
+cmake_dependent_option(MOLD_USE_MIMALLOC "Use mimalloc" ON
+    "NOT APPLE;NOT ANDROID" OFF)
+cmake_dependent_option(MOLD_USE_SYSTEM_MIMALLOC "Use system or vendored mimalloc" ON
+    MOLD_USE_MIMALLOC OFF)
+option(MOLD_USE_SYSTEM_TBB "Use system or vendored TBB" ON)
+
+# Setup mimalloc
+if(MOLD_USE_MIMALLOC)
+    if(MOLD_USE_SYSTEM_MIMALLOC)
+        find_package(mimalloc REQUIRED)
+        target_compile_definitions(mimalloc INTERFACE USE_SYSTEM_MIMALLOC)
+    else()
+        function(mold_add_mimalloc)
+            option(MI_BUILD_TESTS "Build test executables" OFF)
+            add_subdirectory(third-party/mimalloc EXCLUDE_FROM_ALL)
+            target_compile_definitions(mimalloc PRIVATE MI_USE_ENVIRON=0)
+        endfunction()
+
+        mold_add_mimalloc()
+    endif()
+endif()
+
+# Setup TBB
+if(MOLD_USE_SYSTEM_TBB)
+    find_package(TBB REQUIRED)
+else()
+    function(mold_add_tbb)
+        set(BUILD_SHARED_LIBS OFF)
+        set(TBB_TEST OFF CACHE INTERNAL "")
+        set(TBB_STRICT OFF CACHE INTERNAL "")
+        add_subdirectory(third-party/tbb EXCLUDE_FROM_ALL)
+        target_compile_definitions(tbb PRIVATE __TBB_DYNAMIC_LOAD_ENABLED=0)
+    endfunction()
+
+    mold_add_tbb()
+endif()
+
+include(CheckPIESupported)
+check_pie_supported()
+
+add_library(mold-wrapper SHARED)
+set_target_properties(mold-wrapper PROPERTIES
+    POSITION_INDEPENDENT_CODE ON
+    PREFIX "") # Remove the default `lib` prefix
```

Comment:
```suggestion
add_library(mold-wrapper SHARED)
# Remove the default `lib` prefix
set_property(TARGET mold-wrapper PROPERTY PREFIX "")
```

Okay, one last thing. According to https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html

> This property is `True` by default for `SHARED` and `MODULE` library targets and `False` otherwise.

You also weren't checking the output either way, like in the [example](https://cmake.org/cmake/help/latest/module/CheckPIESupported.html#examples).